core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -100,7 +101,20 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
       val parameterTaskPrevPageSize = request.getParameter("task.prevPageSize")
 
       val taskPage = Option(parameterTaskPage).map(_.toInt).getOrElse(1)
-      val taskSortColumn = Option(parameterTaskSortColumn).getOrElse("Index")
+      val taskSortColumn = Option(parameterTaskSortColumn).map { sortColumn =>
+          // SPARK-12708
+          // Due to YARN-2844, "/" in the url will be encoded to "%252F" when
+          // running in yarn mode. `request.getParameter("task.sort")` will return
+          // "%252F". Therefore we need to decode it until we get the real column name.
+          // SPARK-4313 is similar to this issue.
+          var column = sortColumn
```

Comment:
Make sense. 
@yoshidakuy can you add a new method to `UIUtils` to reduce the similar logic.
Then, call it from `ExecutorThreadDumpPage` and `StagePage`.
Also, could you add a test case for the new method to `UIUtilsSuite`?